Sources/SKSupport/LineTable.swift

```diff
@@ -11,130 +11,59 @@
 //===----------------------------------------------------------------------===//
 
 public struct LineTable: Hashable {
-
-  public struct Line: Hashable {
-
-    /// The zero-based line number.
-    public var index: Int
-
-    /// The UTF-8 byte offset of the start of the line.
-    public var utf8Offset: Int
-
-    /// The UTF-16 code-unit offset of the start of the line.
-    public var utf16Offset: Int { return content.startIndex.encodedOffset }
-
-    /// The content of the line, including the newline.
-    public var content: Substring
-
-    @inlinable
-    public init(index: Int, utf8Offset: Int, content: Substring) {
-      self.index = index
-      self.utf8Offset = utf8Offset
-      self.content = content
-    }
-  }
-
   @usableFromInline
-  struct LineData: Hashable {
-    @usableFromInline
-    var stringIndex: String.Index
-    @usableFromInline
-    var utf8Offset: Int
-  }
-
-  @usableFromInline
-  var impl: [LineData]
+  var impl: [String.Index]
 
   public var content: String
 
   public init(_ string: String) {
     content = string
 
-    if content.isEmpty {
-      impl = [LineData(stringIndex: content.startIndex, utf8Offset: 0)]
-      return
-    }
-
     var i = string.startIndex
-    var utf8Offset = 0
-    var prevUTF16: UInt16 = 0
-
-    impl = [LineData(stringIndex: i, utf8Offset: utf8Offset)]
-
-    let utf16 = string.utf16
-
+    impl = [i]
     while i != string.endIndex {
-      let next = utf16.index(after: i)
-
-      let c = utf16[i]
-      utf8Offset += _utf8Count(c, prev: prevUTF16)
-      prevUTF16 = c
-
-      if c == /*newline*/10 {
-        impl.append(LineData(stringIndex: next, utf8Offset: utf8Offset))
+      let c = string[i]
+      string.formIndex(after: &i)
+      if c == "\n" || c == "\r\n" || c == "\r" {
```

Comment:
@benlangmuir Do you know what's the definition of "newline" in LSP?

Reply:
Your code looks correct.

```ts
export const EOL: string[] = ['\n', '\r\n', '\r'];
```